io/zenoh-links/zenoh-link-tcp/src/unicast.rs

```diff
@@ -212,15 +213,23 @@ impl LinkManagerUnicastTcp {
             .peer_addr()
             .map_err(|e| zerror!("{}: {}", dst_addr, e))?;
 
+        zenoh_util::net::set_bind_to_device_tcp_stream(&stream, iface);
+
         Ok((stream, src_addr, dst_addr))
     }
 
-    async fn new_listener_inner(&self, addr: &SocketAddr) -> ZResult<(TcpListener, SocketAddr)> {
+    async fn new_listener_inner(
+        &self,
+        addr: &SocketAddr,
+        iface: &Option<String>,
```

Comment:
In general is always better to pass `Option<&T>` rather than `&Option<T>`.
You can read more about the rationale behind [here](https://www.reddit.com/r/rust/comments/wqsxk2/is_it_better_to_pass_optiont_or_optiont/).

Reply:
Good to know! Thanks!